flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -668,6 +678,37 @@ public int getTotalNumberOfVertices() {
         };
     }
 
+    public EdgeManager getEdgeManager() {
+        return edgeManager;
+    }
+
+    public void registerExecutionVertex(ExecutionVertexID id, ExecutionVertex vertex) {
+        executionVerticesById.put(id, vertex);
+    }
+
+    public void registerResultPartition(
+            IntermediateResultPartitionID id, IntermediateResultPartition partition) {
+
+        resultPartitionsById.put(id, partition);
+    }
+
+    public ExecutionVertex getVertex(ExecutionVertexID id) {
```

Comment:
usually `null` is expected when getting a value from a map. But we do not want it to happen for this.
However, adding `checkNotNull()` everywhere when invoking this method looks ugly.
So I prefer to name it as `getVertexOrThrow` and add a `checkNotNull()` inside to ensure we can find the vertex.